File: app/src/main/java/framework/sql/GreenDaoDbUtils.java
package framework.sql;

import java.util.List;

/**
 * by y on 2016/11/22
 */

public class GreenDaoDbUtils {
    public static SearchSuffixDao getSuffixDb() {
        return GreenDaoUtils.getInstance().getSearchSuffixDao();
    }

    public static List<SearchSuffix> getSuffixAll() {
        return getSuffixDb().loadAll();
    }

    public static boolean isEmpty(String key) {
        return getSuffixDb().queryBuilder().where(SearchSuffixDao.Properties.Suffix.eq(key)).unique() == null;
    }

    public static void clear(int key) {
        getSuffixDb().deleteByKey(key);
    }

    public static void clear() {
        getSuffixDb().deleteAll();
    }

}

File: app/src/main/java/framework/sql/sqlutil/GreenDaoDbUtils.kt
package framework.sql.sqlutil

import framework.sql.SearchSuffix
import framework.sql.SearchSuffixDao

/**
 * by y on 2016/11/22
 */

object GreenDaoDbUtils {
    val suffixDb: SearchSuffixDao
        get() = GreenDaoUtils.instance.searchSuffixDao

    val suffixAll: List<SearchSuffix>
        get() = suffixDb.loadAll()

    fun isEmpty(key: String): Boolean {
        return suffixDb.queryBuilder().where(SearchSuffixDao.Properties.Suffix.eq(key)).unique() == null
    }

    fun clear(key: Int) {
        suffixDb.deleteByKey(key)
    }

    fun clear() {
        suffixDb.deleteAll()
    }

}
